lib/chef/provider/package.rb

```diff
@@ -218,6 +218,28 @@ def action_reconfig
         end
       end
 
+      def action_lock
+        multipackage_api_adapter(@current_resource.package_name, @new_resource.version) do |name, version|
+          if package_locked(name, version) == true
+            Chef::Log.debug("#{new_resource} is already locked")
+          else
+            lock_package(name, version)
+            Chef::Log.info("#{@new_resource} locked")
+          end
```

Comment:
Thanks for that, didn't notice it!
Added the converge_by... formatter output and why-run are now also working.
In additon to that I added more tests, added yum and fixed some issues I noticed.